lib/ansible/modules/windows/win_xml.ps1

```diff
@@ -0,0 +1,191 @@
+        if ($expected.InnerText -ne $actual.InnerText) {
+           throw "inner text mismatch for actual=" + $actual.Name
+        }
+    }
+    elseif ($actual.InnerText) {
+        throw "actual has inner text but expected does not for actual=" + $actual.Name
+    }
+}
+
+function FindElementsByName($node, $name) {
+    $elements = @()
+    foreach ($n in $node.ChildNodes) {
+        if ($n.Name -eq $name) {
+            $elements += $n
+        }
+    }
+    return ,$elements
+}
+
+function BackupFile($path) {
+	$backuppath = $path + "." + [DateTime]::Now.ToString("yyyyMMdd-HHmmss");
+	Copy-Item $path $backuppath;
+	return $backuppath;
+}
+
+$params = Parse-Args $args -supports_check_mode $true
+$dest = Get-AnsibleParam $params "path" -type "path" -FailIfEmpty $true -aliases "dest", "file"
+$fragment = Get-AnsibleParam $params "fragment" -type "str" -FailIfEmpty $true -aliases "xmlstring"
+$root = Get-AnsibleParam $params "root" -type "str" -FailIfEmpty $false -Default "DocumentElement" -aliases "xpath"
+$backup = Get-AnsibleParam $params "backup" -type "bool" -FailIfEmpty $false -Default "no" -ValidateSet "no", "yes"
+$type = Get-AnsibleParam $params "type" -type "str" -FailIfEmpty $false -Default "element" -ValidateSet "element", "attribute", "text"
+$attribute = Get-AnsibleParam $params "attribute" -type "str" -FailIfEmpty ($type -eq "attribute")
+
+$result = New-Object PSObject @{
+    win_xml = New-Object PSObject
+    changed = $FALSE
+}
+
+If (-Not (Test-Path -Path $dest -PathType Leaf)){
+    Fail-Json $result "Specified path $dest does not exist or is not a file."
+}
+
+[xml]$xmlorig = Get-Content -Path $dest
+$xmlnew = $xmlorig.Clone()
+if ($type -eq "element") {
+    $xmlchild = [xml]$fragment
+
+    $child = $xmlnew.CreateElement($xmlchild.DocumentElement.Name, $xmlnew.$root.NamespaceURI)
+    foreach ($attr in $xmlchild.DocumentElement.Attributes) {
+        $child.SetAttribute($attr);
+    }
+
+    foreach ($node in $xmlchild.DocumentElement.ChildNodes) {
+        $newnode = $xmlnew.CreateElement($node.Name, $xmlnew.$root.NamespaceURI)
+        foreach ($attr in $node.Attributes) {
+            $newnode.SetAttribute($attr)
+        }
+        $newnode.InnerText = $node.InnerText
+        $child.AppendChild($newnode) | Out-Null
+    }
+
+    $elements = FindElementsByName $xmlnew.$root $xmlchild.DocumentElement.Name
+
+    [bool]$add = $TRUE
+    if ($elements.Count) {
+        foreach ($element in $elements) {
+            try {
+                Compare-XmlDocs $xmlchild.DocumentElement $element
+                $add = $FALSE
+                break
+            } catch {
+                $_.InvocationInfo | Out-File c:\win_xml.log
+            }
+        }
+        if ($add) {
+            [void]$xmlnew.$root.AppendChild($child)
+        }
+    } else {
+        [void]$xmlnew.$root.AppendChild($child)
+    }
+
+    if ($add) {
+        $result.changed = $TRUE
+        if (-Not $params._ansible_check_mode) {
```

Comment:
Please add check_mode the same way we do for other modules, using Get-AnsibleParam.
Also, where possible, make use of `-WhatIf` support, and add `-WhatIf` support to the backup function, so we can use that like `-WhatIf:$check_mode` as we tend to do for most cmdlets.